File: scam-notifier-ts/src/storage.ts
import { fetchJwt } from "forta-agent";
import { readFileSync } from "fs";
import * as dotenv from "dotenv";
dotenv.config();

const hasLocalNode = process.env.hasOwnProperty("LOCAL_NODE");
const ownerDB = "https://research.forta.network/database/owner/";

export type secrets = {
  apiKeys: {
    scamNotifier: {
      NEO4J_USERNAME: string;
      PROD_NEO4J_URI: string;
      PROD_NEO4J_PASSWORD: string;
      TEST_NEO4J_URI: string;
      TEST_NEO4J_PASSWORD: string;
      ETHERSCAN_KEY: string;
      POLYGONSCAN_KEY: string;
      BSCSCAN_KEY: string;
    };
  };
};

const getToken = async () => {
  const tk = await fetchJwt({});
  return { Authorization: `Bearer ${tk}` };
};

const loadJson = async (key: string): Promise<object> => {
  if (hasLocalNode) {
    const data = readFileSync("secrets.json", "utf8");
    return JSON.parse(data);
  } else {
    try {
      const response = await fetch(`${ownerDB}${key}`, {
        headers: await getToken(),
      });
      if (response.ok) {
        return await response.json();
      } else {
        throw new Error(
          `Error loading JSON from owner db: ${response.status}, ${response.statusText}`
        );
      }
    } catch (error) {
      throw new Error(`Error loading JSON from owner db: ${error}`);
    }
  }
};

export const getSecrets = async (): Promise<secrets> => {
  return (await loadJson("secrets.json")) as secrets;
};

File: social-eng-contract-py/src/storage.py
import json
import os
import aiohttp

import forta_bot

owner_db = "https://research.forta.network/database/owner/"

test_mode = "main" if 'FORTA_ENV' in os.environ and 'production' in os.environ.get(
    'FORTA_ENV') else "test"

async def _token():
    tk = await forta_bot.fetch_jwt()
    return {"Authorization": f"Bearer {tk}"}

async def _load_json(key: str) -> object:
    if test_mode == "test":
        with open("secrets.json") as f:
            return json.load(f)
    else:
        async with aiohttp.ClientSession() as session:
            async with session.get(f"{owner_db}{key}", headers=await _token()) as res:
                if res.status_code == 200:
                    return await res.json()
                else:
                    raise Exception(
                        f"error loading json from owner db: {res.status_code}, {res.text}")

async def get_secrets():
    return await _load_json("secrets.json")
